apps/fluent-tester/src/TestComponents/Switch/E2ESwitchTest.tsx

```diff
@@ -0,0 +1,22 @@
+import * as React from 'react';
+import { Stack } from '@fluentui-react-native/stack';
+import { Switch } from '@fluentui-react-native/switch';
+import { Text } from 'react-native';
+import { stackStyle } from '../Common/styles';
+import { SWITCH_TEST_COMPONENT, SWITCH_TOGGLE_ON, SWITCH_TOGGLE_OFF } from './consts';
+
+export const E2ESwitchTest: React.FunctionComponent = () => {
+  const [showTextChecked, setShowChecked] = React.useState(false);
+
+  const onToggleChecked = (checked: boolean) => {
+    setShowChecked(!checked);
+  };
+
+  return (
+    <Stack style={stackStyle}>
+      <Switch testID={SWITCH_TEST_COMPONENT} label={'Checked'} checked={showTextChecked} onChange={onToggleChecked} />
+      {showTextChecked && <Text testID={SWITCH_TOGGLE_ON}>Checked Switch toggled to true</Text>}
```

Comment:
I'd change a couple things about this logic:

The goal of showing this text is not to test the checked state of the Toggle; the goal is to validate that the `onChange()` callback was fired. Ideally, we'd be able to spy on the onChange() function and determine when it's been called, but this UI Automation doesn't have that capability.

Because of that, we have to use this hacky-ish workaround to show text if the onChange() callback was fired.

Currently, it looks like this logic tests for the state of the `checked` prop for Toggle, which is not the goal (that's very easily done in a different way). The goal is to check if the onChange() was fired.

Please look at how Checkbox does this (same exact logic for onChange()). There should only be one constant `SWITCH_ON_PRESS`.

Relevant files + methods:
CheckboxPageObject.ts - didOnChangeCallbackFire()
Checkbox.spec.win.ts -   it("Click on a Checkbox -> Validate it toggles correctly AND calls the user's onChange")`
CheckboxE2ETest.tsx - `{checkboxPressed ? <Text testID={CHECKBOX_ON_PRESS}>Checkbox Selected</Text> : null}`

If you have any further questions, or if this is confusing, feel free to reach out! It's very simple and might be confusing over text, so a call might help if you're confused

Reply:
I've implemented testing onChange like you've described. Thanks for writing out all this info, it was really helpful! :)